regparser/grammar/utils.py

```diff
@@ -117,6 +125,9 @@ def initial_regex(grammar):
         if (isinstance(grammar, pyparsing.And) and
                 isinstance(grammar.exprs[0], pyparsing.Optional)):
             return recurse(grammar.exprs[0].expr) | recurse(grammar.exprs[1])
+        if (isinstance(grammar, pyparsing.And) and
+                isinstance(grammar.exprs[0], pyparsing.Empty)):
+            return recurse(grammar.exprs[1])
```

Comment:
Are these changes related to the `keep_pos`/`getTokensEndLoc` switch, or is this addressing a separate issue?